app/configurator/interactive-filters/time-slider.tsx

```diff
@@ -169,14 +191,49 @@ const Slider = () => {
   });
 
   React.useEffect(() => {
-    // Dispatch time slider update event when progress changes.
-    if (IFState.timeSlider.value?.getTime() !== timeline.value) {
+    if (IFState.timeSlider.type !== timeline.type) {
       dispatch({
         type: "SET_TIME_SLIDER_FILTER",
-        value: new Date(timeline.value),
+        value: {
+          type: timeline.type,
+          value: undefined,
+        },
+      });
+    }
+  }, [IFState.timeSlider.type, timeline.type, dispatch]);
+
+  React.useEffect(() => {
+    if (
+      IFState.timeSlider.type === "interval" &&
+      IFState.timeSlider.value?.getTime() !== timeline.value
+    ) {
+      dispatch({
+        type: "SET_TIME_SLIDER_FILTER",
+        value: {
+          type: "interval",
+          value: new Date(timeline.value),
+        },
+      });
+    }
+
+    if (
+      IFState.timeSlider.type === "ordinal" &&
+      IFState.timeSlider?.value !== timeline.value
+    ) {
+      dispatch({
+        type: "SET_TIME_SLIDER_FILTER",
+        value: {
+          type: "ordinal",
+          value: timeline.value as string,
+        },
       });
     }
-  }, [timeline.value, IFState.timeSlider.value, dispatch]);
+  }, [
+    timeline.value,
+    IFState.timeSlider.type,
+    IFState.timeSlider.value,
+    dispatch,
+  ]);
```

Comment:
Why do we need an useEffet here ? Couldn't the dispatch be done in an event handler ? Or directly in the reducer ?

Reply:
Hmmm, I think it can't be done in event handler, as this `useEffect` is also called when timeline is playing (and the `onChange` event only happens when user interacts with the timeline) :(

Or did you mean something else with the reducer? I think we wouldn't have access to timeline in the InteractiveFilters reducer 🤔